cli/src/cmd/bind.rs

```diff
@@ -0,0 +1,184 @@
+        long = "bindings-root",
+        value_hint = ValueHint::DirPath
+    )]
+    #[serde(skip)]
+    pub bindings_root: Option<PathBuf>,
+
+    #[clap(
+        long = "crate-name",
+        help = "The name of the rust crate to generate. This should be a valid crates.io crate name. However, it is not currently validated by this command."
+    )]
+    #[serde(skip)]
+    crate_name: Option<String>,
+
+    #[clap(
+        long = "crate-version",
+        help = "The version of the rust crate to generate. This should be a standard semver version string. However, it is not currently validated by this command."
+    )]
+    #[serde(skip)]
+    crate_version: Option<String>,
+
+    #[clap(
+        long = "overwrite",
+        help = "Overwrite existing generated bindings. If set to false, the command will check that the bindings are correct, and then exit. If set to true, it will instead delete and overwrite the bindings."
+    )]
+    #[serde(skip)]
+    overwrite: bool,
+
+    #[clap(long = "single-file", help = "Generate bindings as a single file.")]
+    #[serde(skip)]
+    single_file: bool,
+}
+
+impl BindArgs {
+    /// Get the path to the foundry artifacts directory
+    fn artifacts(&self) -> PathBuf {
+        let c: Config = self.into();
+        c.out
+    }
+
+    /// Get the path to the root of the autogenerated crate
+    fn bindings_root(&self) -> PathBuf {
+        self.bindings_root.clone().unwrap_or_else(|| self.artifacts().join("bindings"))
+    }
+
+    /// Get the name of the autogenerated crate
+    fn crate_name(&self) -> String {
+        self.crate_name.clone().unwrap_or_else(|| "foundry-contracts".to_owned())
+    }
+
+    /// Get the version of the autogenerated crate
+    fn crate_version(&self) -> String {
+        self.crate_version.clone().unwrap_or_else(|| "0.0.1".to_owned())
+    }
+
+    /// `true` if the arguments set `crate_version` or `crate_name`
+    fn gen_crate(&self) -> bool {
+        self.crate_name.is_some() || self.crate_version.is_some()
+    }
+
+    /// `true` if the bindings root already exists
+    fn bindings_exist(&self) -> bool {
+        self.bindings_root().is_dir()
+    }
+
+    /// Instantiate the multi-abigen
+    fn get_multi(&self) -> eyre::Result<MultiAbigen> {
+        MultiAbigen::from_json_files(self.artifacts()).map_err(eyre::Report::msg)
+    }
+
+    /// Check that the existing bindings match the expected abigen output
+    fn check_existing_bindings(&self) -> eyre::Result<()> {
+        let bindings = self.get_multi()?.build().map_err(eyre::Report::msg)?;
+        println!("Checkign bindings for {} contracts", bindings.len());
+        if self.gen_crate() {
+            bindings
+                .ensure_consistent_crate(
+                    self.crate_name(),
+                    self.crate_version(),
+                    self.bindings_root(),
+                    self.single_file,
+                )
+                .map_err(eyre::Report::msg)?;
```

Comment:
this shouldn't be needed once we change abigen to use eyre, but not a blocker for merging